src/ApiCore/GapicClientTrait.php

```diff
@@ -86,18 +86,26 @@ public function close()
     private static function getGapicVersion(array $options)
     {
         if (!self::$gapicVersion) {
-            if (isset($options['versionFile']) && file_exists($options['versionFile'])) {
-                self::$gapicVersion = trim(file_get_contents(
-                    $options['versionFile']
-                ));
-            } elseif (isset($options['libVersion'])) {
-                self::$gapicVersion = $options['libVersion'];
-            }
+            self::$gapicVersion = isset($options['libVersion'])
+                ? $options['libVersion']
+                : self::getVersionFileContents();
         }
 
         return self::$gapicVersion;
     }
 
+    private static function getVersionFileContents()
+    {
+        $clientFile = (new \ReflectionClass(__CLASS__))->getFileName();
+        $versionFile = substr(
+            $clientFile,
+            0,
+            strrpos($clientFile, '/src/') ?: strrpos($clientFile, '\\src\\')
+        ) . DIRECTORY_SEPARATOR . 'VERSION';
+
+        return @file_get_contents($versionFile) ?: null;
```

Comment:
Is it safe to assume that if the VERSION file doesn't exist that the component is simply unreleased? If so, perhaps `dev` or some other indicator of a pre-release component would be preferable to `null`?

Reply:
I don't believe that will always be safe to assume, as this may be run outside the context of google-cloud-php.